cli/src/main/java/org/jboss/as/cli/impl/aesh/cmd/deployment/AbstractUndeployCommand.java

```diff
@@ -179,10 +179,8 @@ private static ModelNode buildRequest(CommandContext ctx, String name,
                     serverGroups = Arrays.asList(serverGroupsStr.split(","));
                 }
 
-                if (serverGroups.isEmpty()) {
-                    if (keepContent) {
-                        throw new OperationFormatException("None of the server groups is specified or references specified deployment.");
-                    }
+                if (serverGroups.isEmpty() && keepContent && !isDisableAll()) {
```

Comment:
I would move this check in the if allRelevantServerGroups is true block.

Reply:
I don't think this is possible, as this will be used also in case when the user won't specify any group

Reply:
I would not remove the if(keepContent). I sketched a diff that I will send you directly, better than multiple comments ;-).